plugins/backend/backstage-plugin-argo-cd-backend/src/service/argocd.service.ts

```diff
@@ -978,16 +1043,90 @@ export class ArgoService implements ArgoServiceApi {
     };
 
     let statusText: string = '';
+    try {
+      const response: GetArgoApplicationFetchResponse = (await fetch(
+        `${url}/api/v1/applications/${argoApplicationName}`,
+        options,
+      )) as GetArgoApplicationFetchResponse;
+
+      statusText = response.statusText;
+
+      if (response.status === 200) {
+        return { ...(await response.json()), statusCode: response.status };
+      }
+      return { ...(await response.json()), statusCode: response.status };
+    } catch (error) {
+      this.logger.error(
+        `Error Getting Argo Application Information For Argo Instance Name ${
+          argoInstanceName || url
+        } - searching for application ${argoApplicationName} - ${JSON.stringify(
+          { statusText, error: (error as Error).message },
+        )}`,
+      );
+      throw error;
+    }
+  }
+
+  // @see https://cd.apps.argoproj.io/swagger-ui#operation/ApplicationService_TerminateOperation
+  async terminateArgoAppOperation({
+    argoAppName,
+    argoInstanceName,
+    baseUrl,
+    argoToken,
+  }: {
+    argoAppName: string;
+    argoInstanceName?: string;
```

Comment:
These again can be non-optional if they are actually required

Reply:
I will fix the type to be accurate.  Thank you!